.github/workflows/cd.yml

```diff
@@ -9,35 +8,63 @@ on:
   release:
     types:
       - published
-
 concurrency:
-  group: ${{ github.workflow }}-${{ github.ref }}
+  group: "${{ github.workflow }}-${{ github.ref }}"
   cancel-in-progress: true
-
 env:
   FORCE_COLOR: 3
-
 jobs:
   dist:
     name: Distribution build
     runs-on: ubuntu-latest
-
     steps:
       - uses: actions/checkout@v4
         with:
           fetch-depth: 0
 
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.11
+
+      - name: Install dependencies
+        run: |
+          pip install requests==2.31.0 pandas==2.1.1 \
+          google-cloud-bigquery==3.12.0 pyarrow==13.0.0 \
+          db-dtypes==1.1.1 pygithub==2.1.1
+        shell: bash
+
+      - name: Authorize Google Cloud
+        uses: google-github-actions/auth@v1
+        with:
+          credentials_json: "${{ secrets.SERVICE_ACCOUNT_KEY }}"
+          create_credentials_file: true
+          export_environment_variables: true
+
+      - name: Run script to get the latest idc index
+        run: |
+          python scripts/python/idc_index_builder.py
+        shell: bash
+        env:
+          GCP_PROJECT_ID: "${{ secrets.GCP_PROJECT_ID }}"
+
       - uses: hynek/build-and-inspect-python-package@v2
 
+      - name: Upload SQL queries
+        uses: actions/upload-artifact@v4
+        with:
+          name: sql-queries
+          path: scripts/sql
```

Comment:
```suggestion
```

After switching to having `QUERY_TEMPLATE_GENERATE_IDC_INDEX`, this could be removed